lib/compilers/rust.ts

```diff
@@ -52,11 +55,54 @@ export class RustCompiler extends BaseCompiler {
         this.linker = this.compilerProps('linker');
     }
 
-    getSharedLibraryPathsAsArguments() {
+    override getSharedLibraryPathsAsArguments(libraries, libDownloadPath) {
         return [];
     }
 
-    optionsForBackend(backendOptions, outputFilename) {
+    override getSharedLibraryLinks(libraries): string[] {
+        return [];
+    }
+
+    override getIncludeArguments(libraries) {
+        const includeFlag = '--extern';
+        return _.flatten(
+            _.map(libraries, selectedLib => {
+                const foundVersion = this.findLibVersion(selectedLib);
+                if (!foundVersion) return false;
+                const list: string[] = [];
+                for (const rlib of foundVersion.path) {
+                    list.push(includeFlag, `${foundVersion.name}=${foundVersion.name}/build/debug/${rlib}`);
+                }
+                return list;
+            }),
+        );
+    }
+
+    override orderArguments(
+        options,
+        inputFilename,
+        libIncludes,
+        libOptions,
+        libPaths,
+        libLinks,
+        userOptions,
+        staticLibLinks,
+    ) {
+        return options.concat(userOptions, libIncludes, libOptions, libPaths, libLinks, staticLibLinks, [
+            this.filename(inputFilename),
+        ]);
+    }
+
+    override async setupBuildEnvironment(key, dirPath): Promise<BuildEnvDownloadInfo[]> {
```

Comment:
Override of the setupBuildEnvironment option, because the original checks for binary/execution mode and for Rust we Always need to download the libraries